app/models/task.rb

```diff
@@ -5,9 +5,17 @@ class Task < ApplicationRecord
     state :unstarted, initial: true
     state :in_progress
     state :completed
+    state :correcting
 
     event :completed do
-      transitions from: %i[unstarted in_progress completed], to: :completed
+      transitions from: %i[unstarted in_progress correcting completed], to: :completed
+    end
+
+    event :cancel_correction do
```

Comment:
I think some model spec coverage on this new event would be a good idea.

Reply:
I've added a spec to task_spec